app/services/hyrax/resource_characterizer.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # Characterizes a Valkyrie based FileSet
+  class ResourceCharacterizer
+    ##
+    # @!attribute [rw] source
+    #   @return [#characterize]
```

Comment:
Same as above. I would expect `source` to return an instance of a Valkyrie-based file set.